### PR TITLE
[Libraries] Refactor File_Upload Class

### PR DESCRIPTION
**Pull request by johnsaigle**
# Refactor of File Upload

## Purpose/Background

- It is important to have a centralized way of doing File Uploads, especially in PHP, as it is a major security concern. 

- Different modules have implemented ad-hoc file uploaders ranging from fairly secure to very vulnerable (`genomic_browser`, `imaging_uploader`, `document_repository`, `media`, others?)

- This class needs to be updated to allow for future refactoring of the above modules 

- This class is only used by `imaging_uploader` and as such as loosely coupled. This makes it suitable for tweaking and updating

----

## Things this PR (hopefully) does...

- [x] Removes unused flags and functions. See the differences in the `__construct` function for the deleted flags.

- [x] Changes some variable names to be more descriptive

- [x] Introduces minor refactoring to improve readability and reduce redundancy

- [x] Specifies the scope of functions to make code more clear and assist in further refactoring

- [x] Suggests further refactoring efforts with references to other code (mostly `imaging_uploader`)

- [x] Improves error messages, code comments, and adjusts programming conventions to be more consistent with more recent LORIS coding conventions

## Things this PR (hopefully) does not do....

- [x] Affect existing functionality in `imaging_uploader` -- it's not used anywhere else

### CODE REVIEW

**Comment by johnsaigle** [comment]
`data_release` and `instrument_manager` also have their own file uploaders...

File: php/libraries/File_Upload.class.inc [hunk 1]
```diff
@@ -508,144 +465,6 @@ class File_Upload
         return $new_name;
     }
 
-     /**
```
Comment (ridz1208) [inline php/libraries/File_Upload.class.inc]: removing the following functions seems to be removing some renaming options

@driusan @MounaSafiHarab either of you know why they were added in the first place ?? seems like imaging upploader is the only module that uses this class

File: php/libraries/File_Upload.class.inc [hunk 2]
```diff
@@ -24,25 +25,31 @@
...
+
     /**
      * Construct the File_Upload class. This sets the default parameters
      * for a file which can be overwritten by calling the set* methods
      */
-    function __construct()
+    public function __construct()
     {
-        //Set Defaults
-        $this->renameMode        = "incremental";
```
Comment (ridz1208) [inline php/libraries/File_Upload.class.inc]: @MounaSafiHarab to your knowledge does anyone have this value set to another option then incremental ??

**Comment by xlecours** [comment]
Yes! Interface and decoupled from Loris_Form.
Here is a good interface to use: http://www.php-fig.org/psr/psr-7/#36-psrhttpmessageuploadedfileinterface

**Comment by nicolasbrossard** [comment]
We'll leave this PR open for now and I'll implement the changes I requested in another PR (coming soon...)

**Comment by ridz1208** [comment]
closing in favour of https://github.com/aces/Loris/pull/3476

**Pull request closed without merging**